debezium-core/src/test/java/io/debezium/relational/history/JdbcDatabaseHistoryTest.java

```diff
@@ -0,0 +1,126 @@
+                    .name("first")
+                    .jdbcType(Types.VARCHAR)
+                    .type("VARCHAR")
+                    .length(22)
+                    .optional(false)
+                    .create())
+            .setPrimaryKeyNames("first")
+            .create();
+    TableChanges tableChanges = new TableChanges().create(table);
+    HistoryRecord historyRecord = new HistoryRecord(source, position, databaseName, schemaName, ddl, tableChanges);
+    Map<String, Object> position2 = Collect.linkMapOf("file", "x.log", "positionInt", 100, "positionLong", Long.MAX_VALUE, "entry", 2);
+    TableId tableId2 = new TableId(databaseName, schemaName, "bar");
+    Table table2 = Table.editor()
+            .tableId(tableId2)
+            .addColumn(Column.editor()
+                    .name("first")
+                    .jdbcType(Types.VARCHAR)
+                    .type("VARCHAR")
+                    .length(22)
+                    .optional(false)
+                    .create())
+            .setPrimaryKeyNames("first")
+            .create();
+    TableChanges tableChanges2 = new TableChanges().create(table2);
+    HistoryRecord historyRecord2 = new HistoryRecord(source, position, databaseName, schemaName, ddl, tableChanges2);
+    DdlParser ddlParser = new TestAntlrDdlParser();
+
+    @Before
+    public void beforeEach() {
+        history = new JdbcDatabaseHistory();
+        history.configure(Configuration.create()
+                .with(JDBC_URI, "jdbc:sqlite:" + dbFile)
+                .with(JDBC_USER, "user")
+                .with(JDBC_PASSWORD, "pass")
+                .build(), null, DatabaseHistoryMetrics.NOOP, true);
+        history.start();
+    }
+
+    @After
+    public void afterEach() throws IOException {
+        if (history != null) {
+            history.stop();
+        }
+        Files.delete(Paths.get(dbFile));
+    }
+
+    @Test
+    public void shouldNotFailMultipleInitializeStorage() {
+        history.initializeStorage();
+        history.initializeStorage();
+        history.initializeStorage();
+        assertThat(history.storageExists());
+    }
+
+    @Test
+    public void shouldRecordChangesAndRecover() throws InterruptedException {
+        history.record(source, position, historyRecord.databaseName(), historyRecord.schemaName(), historyRecord.ddl(), tableChanges);
+        history.record(source, position, historyRecord.databaseName(), historyRecord.schemaName(), historyRecord.ddl(), tableChanges);
+        Tables tables = new Tables();
+        history.recover(source, position, tables, ddlParser);
+        assertThat(tables.size() == 1);
+        assertThat(tables.forTable(tableId).equals(table));
+        history.record(source, position2, historyRecord2.databaseName(), historyRecord2.schemaName(), historyRecord2.ddl(), tableChanges2);
+        history.record(source, position2, historyRecord2.databaseName(), historyRecord2.schemaName(), historyRecord2.ddl(), tableChanges2);
```

Comment:
not sure why recording it once not returning `table2`? had to call it twice to make test pass